File: apps/driver-frontend/lib/features/payout_methods/domain/entitites/payout_account.input.dart
import 'package:freezed_annotation/freezed_annotation.dart';

part 'payout_account.input.freezed.dart';

@freezed
class PayoutAccountInput with _$PayoutAccountInput {
  const factory PayoutAccountInput({
    required String name,
    required String payoutMethodId,
    required String accountNumber,
    required String? routingNumber,
    required String bankName,
    required String? accountHolderName,
    required String? branchName,
    required bool isDefault,
    required String? accountHolderAddress,
    required String? accountHolderCity,
    required String? accountHolderState,
    required String? accountHolderZip,
    required String? accountHolderCountry,
    required String? accountHolderPhone,
    required DateTime? accountHolderDateOfBirth,
  }) = _PayoutAccountInput;
}

File: apps/driver-api/src/app/payout/dto/payout-account.input.ts
import { Field, ID, InputType } from '@nestjs/graphql';

@InputType()
export class PayoutAccountInput {
  name!: string;
  @Field(() => ID)
  payoutMethodId!: number;
  accountNumber!: string;
  routingNumber?: string;
  bankName!: string;
  accountHolderName?: string;
  branchName?: string;
  isDefault!: boolean;
  accountHolderAddress?: string;
  accountHolderCity?: string;
  accountHolderState?: string;
  accountHolderZip?: string;
  accountHolderCountry?: string;
  accountHolderPhone?: string;
  accountHolderDateOfBirth?: Date;
}
